zoo/zoo.py

```diff
@@ -0,0 +1,281 @@
+
+        moved_animal.cage = cage_to
+
+        for animal in move_to.cage_contents:
+            self.dinner_time(animal_name, animal.name)
+
+        for animal in move_to.cage_contents:
+            self.dinner_time(animal.name, animal_name)
+
+        cage_to.cage_relationships()
+
+        print("{} has been moved from {} to {}".format(
+            moved_animal.name, cage_from, cage_to))
+
+
+class Cage(object):
+    """Creates a ZooCage object with a cage_name.
+
+       Possibility to extend a BaseCage class that would have a 'prefferred' contents. e.g a gazelle cage that
+       prevented gazelles being eaten in their preferred cage as they have hiding places or whatever.
+       """
+
+    instances = 0
+
+    def __init__(self, cage_name, zoo=None):
+        """Cage is created with a name and an empty list for contents."""
+        self.id = self.instances
+        self.name = cage_name
+        self.cage_contents = []
+        if zoo:
+            self.zoo = zoo
+        Cage.instances += 1
+
+    def __del__(self):
+        """Cage can be deleted while keeping track of the total."""
+        Cage.instances -= 1
+
+    def __str__(self):
+        """Human readable string returned when the class object is printed."""
+        return '<%s: %s Animals>' % (self, len(self.cage_contents))
+
+    def __repr__(self):
+        """Class representation, returned when class object is called."""
+        return '<%s: %s (%s)>' % (self.__class__.__name__, self.name, self.cage_contents)
+
+    def add_animal(self, animal):
+        animal.get_competition_score()
+        animal.cage = self
+        self.cage_contents.append(animal)
+        relationship = self.cage_relationships()
+        if relationship:
+            return relationship
+        else:
+            return self
+
+    def cage_relationships(self):
+        """If you put prey and predator in the same cage, then all the prey should be eaten by the predator."""
+        if len(self.cage_contents) > 1:
+            COMPETITIVENESS = 2  # The higher the competitiveness the harder it is to eat animals. Animals try harder to avoid being eaten.
+            competition_scores = [animal.competition_score for animal in self.cage_contents]
+            predators = [animal for animal in self.cage_contents if animal.competition_score == min(competition_scores)]
+            eaten_animals = 0
+
+            for animal in self.cage_contents:
+                if animal.competition_score > min(competition_scores) * COMPETITIVENESS:  # There must be a difference in competitiveness of a certain amount calculated with a CONSTANT
+                    animal.status = animal.STATUS_DEAD
+                    animal.sit_rep = "I was eaten by {}".format(predators)
+                    eaten_animals += 1
+            if eaten_animals:
+                return "{} animals eaten".format(eaten_animals)
+            else:
+                return None
+
+
+class BaseSpecies(object):
```

Comment:
I would say that a better name for this class would be `Animal`.  Species is a characteristic of an animal, but the animal is the class of object.  We can discuss this more.